solidity/test/NetworkSettings.js

```diff
@@ -0,0 +1,87 @@
+const { contract, accounts } = require('@openzeppelin/test-environment');
+const { BN, constants, expectRevert } = require('@openzeppelin/test-helpers');
+const { expect } = require('../../chai-local');
+
+const NetworkSettings = contract.fromArtifact('NetworkSettings');
+
+const PPM_RESOLUTION = new BN(1000000);
+
+describe('NetworkSettings', () => {
+    let networkSettings;
+
+    const owner = accounts[1];
+    const nonOwner = accounts[2];
+
+    const validFeeWallet1 = accounts[3];
+    const validFeeWallet2 = accounts[4];
+
+    const validFeePortion1 = new BN(111);
+    const validFeePortion2 = new BN(222);
+
+    const invalidFeeWallet = constants.ZERO_ADDRESS;
+    const tooSmallFeePortion = new BN(0);
+    const tooLargeFeePortion = PPM_RESOLUTION.addn(1);
+
+    const expectReturn = async (method, object) => {
+        expect(JSON.stringify(await method)).to.be.equal(JSON.stringify(object));
+    };
+
+    it('should revert when creating a contract with an invalid fee wallet', async () => {
+        await expectRevert(NetworkSettings.new(invalidFeeWallet, validFeePortion1), 'ERR_INVALID_ADDRESS');
+    });
+
+    it('should revert when creating a contract with a too-small fee portion', async () => {
+        await expectRevert(NetworkSettings.new(validFeeWallet1, tooSmallFeePortion), 'ERR_INVALID_PORTION');
+    });
+
+    it('should revert when creating a contract with a too-large fee portion', async () => {
+        await expectRevert(NetworkSettings.new(validFeeWallet1, tooLargeFeePortion), 'ERR_INVALID_PORTION');
+    });
+
+    it('should revert when setting an invalid fee wallet', async () => {
+        let networkSettings = await NetworkSettings.new(validFeeWallet1, validFeePortion1);
+        await expectRevert(networkSettings.setFeeWallet(invalidFeeWallet), 'ERR_INVALID_ADDRESS');
+        await expectReturn(networkSettings.feeParams(), {0: validFeeWallet1, 1: validFeePortion1});
```

Comment:
This is a weird syntax, but I do understand why you're doing this. I'd personally compare the returned array explicitly, but it's fine

Reply:
We don't have a unified set of rules for comparisons in our tests, so I just went along with this option...